lib/modules/versioning/unity3d-packages/index.ts

```diff
@@ -0,0 +1,57 @@
+import is from '@sindresorhus/is';
+import { regEx } from '../../../util/regex';
+import type { GenericVersion } from '../generic';
+import { GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'unity3d-packages';
+export const displayName = 'Unity3D Packages';
+export const urls = [
+  'https://docs.unity3d.com/Manual/upm-semver.html',
+  'https://docs.unity3d.com/Manual/upm-lifecycle.html',
+];
+export const supportsRanges = false;
+
+class Unity3dPackagesVersioningApi extends GenericVersioningApi {
+  private static readonly parsingRegex = regEx(
+    /^(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)(-?(?<label>.*))/,
+  );
+  private static readonly unstableRegex = regEx(/^(exp.|pre.|preview.)/);
+
+  protected _parse(version: string): GenericVersion | null {
+    const matches = Unity3dPackagesVersioningApi.parsingRegex.exec(version);
+    if (!matches?.groups) {
+      return null;
+    }
+    const { major, minor, patch, label } = matches.groups;
+
+    const release = [
+      parseInt(major, 10),
+      parseInt(minor, 10),
+      parseInt(patch, 10),
+    ];
+    const isStable = !Unity3dPackagesVersioningApi.unstableRegex.test(label);
+
+    return { release, prerelease: isStable ? undefined : label, suffix: label };
```

Comment:
```suggestion
    return { release, prerelease: isStable ? undefined : label, suffix: label };
```
do not use prerelease and suffix

Reply:
I don't understand, do you mean don't use both at the same time or don't use either one?

I need the `suffix` for the `_compareOther` method.

And why can't I use `prerelease` even if I don't use it directly? Aren't we supposed to try to fill in as much information as possible so that it can be useful to whoever wants to extend the functionality with custom managers, etc?

Reply:
@viceice can you answer? Thanks

Reply:
they should not have same values. use pre release if the version is unstable. use suffix for some stable compatibility string.

so best is to drop suffix entirely.